client/src/pages/BookingPage.tsx

```diff
@@ -29,49 +32,123 @@ const BookingPage: FC = () => {
       setOpen(true);
     }
   };
+  const isDeskSelected =() =>{
+    return selectedDesk === ""
+  }
+  const isEndTimeAfterStart = () =>{
+    const unixStartTime = getUnixTime(dateValue, startTimeValue);
+    const unixEndTime = getUnixTime(dateValue, endtimeValue);
+    return unixEndTime - unixStartTime <= 0;
+  }
+  const isFutureTime = () =>{
+    const unixStartTime = getUnixTime(dateValue, startTimeValue);
+    return unixStartTime - (Date.now() + UNIX_HOUR) / 1000 <= 0;
+  }
 
-  const [startTimeValue, setStartTimeValue] = useState<Date>(new Date());
-  const [endtimeValue, setEndTimeValue] = useState<Date>(new Date());
-  const [dateValue, setDateValue] = useState<Date>(new Date());
+  const checkFields = () => {
+    return isDeskSelected() || isEndTimeAfterStart() || isFutureTime();
+  };
+
+  const initialStartTime = new Date();
+  initialStartTime.setHours(9, 0, 0, 0);
+  const initialEndTime = new Date();
+  initialEndTime.setHours(17, 0, 0, 0);
+  const [activeStep, setActiveStep] = useState(0);
+  const [startTimeValue, setStartTimeValue] = useState<Date>(initialStartTime);
+  const [endtimeValue, setEndTimeValue] = useState<Date>(initialEndTime);
+  const [dateValue, setDateValue] = useState<Date>(
+    new Date(Date.now() + UNIX_DAY)
+  );
   const [selectedDesk, setSelectedDesk] = useState("");
   const store = {
     startTime: [startTimeValue, setStartTimeValue],
     endTime: [endtimeValue, setEndTimeValue],
     date: [dateValue, setDateValue],
     desk: [selectedDesk, setSelectedDesk],
+    activeStep: [activeStep, setActiveStep],
   };
+
   return (
     <FormContext.Provider value={store}>
       <div>
-        <Grid container spacing={2}>
-          <Grid item xs={8}>
-            <h3>layout picture</h3>
-            <img src={img} alt="layout img" width={1500} height={600} />
-            <Grid container spacing={2}>
-              <Grid item xs={6}>
-                <DateSetter />
+        <Grid container alignItems={"center"}>
+          <Grid item sm={12} lg={7} alignItems={"center"} textAlign={"center"}>
+            <Typography variant="h4">Floor plan</Typography>
+            <img src={img} alt="layout img" width={"100%"} height={"auto"} />
+          </Grid>
+          <Grid item sm={12} lg={5} height={1100}>
+            <Grid container direction={"column"} justifyContent={"center"} paddingTop={5}>
+              <Grid item md={4}>
+              <BookingStepper />
               </Grid>
-              <Grid item xs={6}>
-                <TimeSetter title={"Start time"} type={"Start"} />
-                <TimeSetter title={"End time"} type={"End"} />
+              <Grid item md={6} justifyContent={"center"}>
+              {
```

Comment:
This is a pretty mind-binding construction. I looked into the mui docs and I would suggest to split it up into smaller pieces.

First of all, since the key for the steps is 0, 1, and 2, you can just use an array instead of an object keyed with numbers. They are now strings, but they should be numbers because activeStep is a number.

Then I would move this array outside the component.

Than refactor the contained steps: they can be (small) separate components.

To start you off, it would look something like this:

```
// in a util
// Your current implementations of isFutureTime etc are not pure because they reference dateValue internally
const isFutureTime = (dateValue, startTimeValue) => {
    const unixStartTime = getUnixTime(dateValue, startTimeValue);
    return unixStartTime - (Date.now() + UNIX_HOUR) / 1000 <= 0;
}

// In src/steps/DateStep.tsx
const DateStep: FC = () => {
  const store = {
    startTime: [startTimeValue, setStartTimeValue],
    date: [dateValue, setDateValue],
  };
  return <Box>
      <Typography variant="subtitle1">Please select the date for when you want to book a desk</Typography>
      <DateSetter />
      {isFutureTime(dateValue, startTimeValue) ? (
        <Alert severity="warning">
          Check if the date has passed
        </Alert>
      ) : null}
    </Box>
}

// In src/pages/BookingPage.tsx
const steps = [DateStep, TimeStep, DeskStep];

const BookingPage: FC = () => {
  ...
  return <FormContext...
     ...
     <Grid item md={6} justifyContent={"center"}>
       {steps[activeStep]}
    ...
}
```

Use that context, now we have it!